numpy/__init__.py

```diff
@@ -385,6 +390,14 @@ def __getattr__(attr):
                 f"{__expired_attributes__[attr]}"
             )
 
+        if attr == "chararray":
+            warnings.warn(
+                "`np.chararray` is deprecated and will be removed from "
+                "the main namespace in the future. Use `np.char.chararray` "
+                "instead.", DeprecationWarning, stacklevel=2)
```

Comment:
For the migration, I would say "Use an array with a string or bytes dtype instead," while it will still be available as `np.char.charray`, that will likely also be deprecated before NumPy 2.0 is released.

Reply:
Sure! Updated.